src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -230,6 +230,13 @@ public class AgentKnobs
             new RuntimeKnobSource("ALLOW_TFVC_UNSHELVE_ERRORS"),
             new EnvironmentKnobSource("ALLOW_TFVC_UNSHELVE_ERRORS"),
             new BuiltInDefaultKnobSource("false"));
-    }
 
+        // Set ALLOW_JAVA_CAPABILITY_HIGHER_THAN_9 variable with any value
+        // to disable recognition of Java higher than 9
+        public static readonly Knob DisableRecognitionOfJDKHigherThen9 = new Knob(
+            nameof(DisableRecognitionOfJDKHigherThen9),
+            "Allow to recognise SDK higher than 9",
```

Comment:
Could you please describe it more precisely? 'Recognize Java' -> something similar to 'Recognize JDK and JRE >= 9 installed on machine as agent capability. If set to false - recognize only versions < 9'
Typo: recognise -> recognize

Reply:
I agree with the first part of the suggestion 'Recognize JDK and JRE >= 9 installed on the machine as agent capability' , but regarding the second part we don't consider the exact value for this variable, the idea was that if this variable is set (doesn't matter which value) we disabling recognition of new JDR/JRE versions. If the variable is not set recognition of the new versions enabled. I see, we already using knobs in this way for example [TraceVerbose](https://github.com/microsoft/azure-pipelines-agent/blob/15df9f008c27abcf941a11989cdd8100a4d3952a/src/Agent.Sdk/Knob/AgentKnobs.cs#L98).
So I think the part 'If set to false - recognize only versions < 9' will be better to change to 'If the variable is set to any value - recognize only versions < 9'

Reply:
Changed this description